yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -92,26 +93,68 @@ def run_shieldhit(param_dict):
             with open(abs_input_path, 'w') as temp_input_file:
                 temp_input_file.write(input_dict[config_filename])
 
+        print(os.listdir(tmp_output_path))
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
-        isRunOk = runner_obj.run(settings=settings)
-        if not isRunOk:
+        if not runner_obj.run(settings=settings):
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
-        estimator = runner_obj.get_data()
-
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        estimator_list = runner_obj.get_data()
+
+        return convert_output(estimator_list)
+
+
+def convert_output(estimator_list):
+    """Function for converint simulation output to dictionary"""
+    if not estimator_list:
+        return {"result": "None"}
+
+    result_dict = {"estimators": []}
+    for est_name in estimator_list:
+
+        estimator_dict = {
+            "name" : est_name,
+            "pages": []}
+        for page in estimator_list[est_name].pages:
+
+            page_dict = {"axis": []}
+            for i in range(5):
+                axis_dict = {
+                    "n": int(page.axis(i).n),
+                    "min_val": float(page.axis(i).min_val),
+                    "max_val": float(page.axis(i).max_val),
+                    "name": str(page.axis(i).name),
+                    "unit": str(page.axis(i).unit),
+                    "binning": str(page.axis(i).binning),
+                    "data": []
```

Comment:
Why don't you extract X axis data as well ?
See `axis.data` property (not to be confused with `page.data` property
https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/axis.py#L40